Code/Editor/CryEditDoc.cpp

```diff
@@ -1004,43 +994,45 @@ bool CCryEditDoc::HasLayerNameConflicts()
 bool CCryEditDoc::DoSaveDocument(const QString& filename, TSaveDocContext& context)
 {
     bool& bSaved = context.bSaved;
-    if (bSaved)
+    if (!bSaved)
```

Comment:
The changes to `DoSaveDocument` make be slightly nervous, not to say the transformation isn't good, but unless this is fixing a bug, the mantra '_if it ain't broke, don't fix it_' usually applies. Something like this (as it's such a core operation) if we're going to change really warrants a test to verify the before/after behavior. This unfortunately is easier said than done (especially with this more legacy code) but might be something to consider. We do have quite a few examples of tests throughout the source base now, this might be some logic we could extract to test in isolation?

Reply:
This change is semantic-preserving since the original code was doing:
```cpp
{
    bool& bSaved = context.bSaved;
    if (bSaved)
    {
       // WAS SAVED STUFF HERE
    }

    return bSaved;
}
```
my change was basically flipping the `if` block to reduce nesting ( hopefully increasing readability ):
```cpp
{
    bool& bSaved = context.bSaved;
    if (!bSaved)
    {
       return bSaved;
    }

    // SAVED STUFF HERE
    return bSaved;
}
```
To be honest I was tempted to replace `bool&` with `bool` as well, assigning to `context.bSaved` as needed :)

Reply:
Cool got you, it does seem pretty safe so it's likely fine, it's just these sorts of transformations (even if on first glance if they seem totally harmless) can sometimes cause things to blow up (ask me how I know 😉) In a perfect world we'd have a test but as this is somewhat legacy and there's no existing infrastructure for it I know it's a tall ask, in more modern areas with better test support we usually are more insistent 🙂